File: python/gpt/p116_nc.py
# Python: list current directory without shell execution
import os
import sys

def main() -> int:
    path = "."  # not user-controlled
    try:
        with os.scandir(path) as it:
            for entry in it:
                print(entry.name)
    except OSError as e:
        print(f"scandir('{path}') failed: {e}", file=sys.stderr)
        return 1
    return 0

if __name__ == "__main__":
    raise SystemExit(main())

File: javascript/gpt/p116_nc.js
// JavaScript (Node.js): list current directory without shell execution
"use strict";

const fs = require("fs/promises");

async function main() {
  const path = "."; // not user-controlled
  try {
    const entries = await fs.readdir(path, { withFileTypes: true });
    for (const ent of entries) {
      process.stdout.write(ent.name + "\n");
    }
  } catch (err) {
    process.stderr.write(`readdir('${path}') failed: ${err?.message ?? String(err)}\n`);
    process.exitCode = 1;
  }
}

main();